src/main/kotlin/com/ampnet/reportservice/service/impl/XlsxServiceImpl.kt

```diff
@@ -0,0 +1,147 @@
+    }
+
+    private fun getUsersWithApprovedDeposit(coop: String): List<UserResponse> {
+        val depositOwners = walletService.getOwnersWithApprovedDeposit(coop)
+        val usersWithDeposit = userService.getAllActiveUsers(coop).usersList
+            .filter { depositOwners.contains(it.uuid) }
+        return mapUserResponse(usersWithDeposit)
+    }
+
+    private fun getUsersWithWallet(coop: String): List<UserResponse> {
+        val activeUsers = userService.getAllActiveUsers(coop)
+        val walletOwners = walletService
+            .getWalletsByOwner(activeUsers.usersList.map { UUID.fromString(it.uuid) })
+            .map { it.owner }
+        val usersWithWallet = activeUsers.usersList.filter { walletOwners.contains(it.uuid) }
+        return mapUserResponse(usersWithWallet)
+    }
+
+    private fun writeHeaderLine(type: XlsxType) {
+        sheet = workbook.createSheet("Users-$type")
+        val font = workbook.createFont().apply {
+            bold = true
+            fontHeight = headerFontSize
+        }
+        val style = workbook.createCellStyle().apply { setFont(font) }
+        val row = sheet.createRow(0)
+        var columnCount = 0
+        createCell(row, columnCount++, "User UUID", style)
+        createCell(row, columnCount++, "E-mail", style)
+        createCell(row, columnCount++, "First Name", style)
+        createCell(row, columnCount++, "Last Name", style)
+        createCell(row, columnCount++, "Authentication Method", style)
+        createCell(row, columnCount, "Registration Date", style)
+    }
+
+    private fun writeDataLines(users: List<UserResponse>) {
+        val font = workbook.createFont().apply { fontHeight = dataFontSize }
+        val style = workbook.createCellStyle().apply { setFont(font) }
+        var rowCount = 1
+        for (user in users) {
+            val row: Row = sheet.createRow(rowCount++)
+            var columnCount = 0
+            createCell(row, columnCount++, user.uuid, style)
+            createCell(row, columnCount++, user.email, style)
+            createCell(row, columnCount++, user.firstName, style)
+            createCell(row, columnCount++, user.lastName, style)
+            createCell(row, columnCount++, user.auth, style)
+            createCell(row, columnCount, dateToString(user.createdAt), style)
+        }
+    }
+
+    private fun createCell(row: Row, columnCount: Int, value: String, style: CellStyle) {
+        sheet.autoSizeColumn(columnCount)
+        row.createCell(columnCount).apply {
+            setCellValue(value)
+            cellStyle = style
+        }
+    }
+
+    private fun dateToString(date: Long): String =
+        date.millisecondsToLocalDateTime().format(DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss"))
+
+    private fun mapUserResponse(users: List<UserExtendedResponse>): List<UserResponse> =
+        users.map {
+            UserResponse.newBuilder().apply {
+                uuid = it.uuid
```

Comment:
Doesn't `it` now refer to the `UserResponse.newBuilder()` instance? This would be more clear if you named the `user` value which comes from `users.map`

Reply:
No, but I will name it to `user`